src/cpu-template-helper/src/template/dump/aarch64.rs

```diff
@@ -10,7 +11,18 @@ pub fn config_to_template(cpu_config: &CpuConfiguration) -> CustomCpuTemplate {
     let mut reg_modifiers: Vec<RegisterModifier> = cpu_config
         .regs
         .iter()
-        .map(|reg| reg_modifier!(reg.id, reg.value))
+        .map(|reg| match reg.size() {
+            RegSize::U32 => {
+                reg_modifier!(reg.id, u128::from(reg.value::<u32, 4>()))
+            }
+            RegSize::U64 => {
+                reg_modifier!(reg.id, u128::from(reg.value::<u64, 8>()))
+            }
+            RegSize::U128 => {
+                reg_modifier!(reg.id, reg.value::<u128, 16>())
+            }
+            _ => unreachable!("Only 32, 64 and 128 bit wide registers are supported"),
```

Comment:
Mh, why are these unreachable? Cant a user just edit a snapshot file to contain the id for a register with width > 128bits, and then this code gets triggers (maybe not here, but in one of the other places where panics are inserted). Can we use error handling here? I think there's some SLA we have to not panic (although we already break that in the virtio stack, I suppose)

Reply:
This code is not run during snapshot restoration.